client/app/src/main/java/com/microsoft/research/karya/injection/RepositoryModule.kt

```diff
@@ -30,7 +31,7 @@ class RepositoryModule {
 
     @Provides
     @Singleton
-    fun provideWorkerRepository(workersAPI: WorkersAPI): WorkerRepository {
-        return WorkerRepository(workersAPI)
+    fun provideWorkerRepository(workerAPI: WorkerAPI, workerDao: WorkerDao): WorkerRepository {
```

Comment:
Can this be done using only constructor injection instead of making a separate module? This is a class that we own

Reply:
We are doing constructor injection here, we added this to a module so that hilt does that injection for us instead of us creating instance of the module whenever we want that class.

Reply:
Oh, I guess it happens automatically with Inject annotation in the constructor if hilt knows how to construct the parameters specified in constructors which in this case it knows?

Reply:
I get it. This is so that we have an instance of WorkerRepository, the issue is not of if WorkerRepository can be created without hilt. Thanks for clearing this out